Adam

essafik opened the pull request.

qalgos committed:
chapter 3.3

essafik committed:
Merge pull request #15 from notalp1/patch-5

chapter 3.3: Quri Parts

essafik committed:
Rename chapter 3.3 to 3.3_Quri_Parts

Rename file

qalgos committed:
Update 3.3_Quri_Parts

essafik committed:
Merge pull request #17 from notalp1/patch-7

Update 3.3_Quri_Parts

essafik committed:
Rename 3.3_Quri_Parts to 3.3_Quri_Parts.ipynb

qalgos committed:
Update 3.3_Quri_Parts.ipynb

qalgos committed:
Update 3.3_Quri_Parts.ipynb

essafik committed:
Merge pull request #18 from notalp1/patch-9

Update 3.3_Quri_Parts.ipynb

essafik committed:
Merge branch 'Adam' into patch-10

essafik committed:
Merge pull request #19 from notalp1/patch-10

Update 3.3_Quri_Parts.ipynb

dchung0741 commented on notebooks/3.3_Quri_Parts.ipynb:
```diff
@@ -0,0 +1,381 @@
+       "`quri-parts-chem`: General concepts, Fermion-qubit mapping etc.<br>",
+       "<b>Library support:</b> <br>",
+       "`quri-parts-openfermion` <br>",
+       "<br>",
```
The information here is not the most up-to-date. I'm working on updating this part in the QURI Parts documentation page. I'll let you know once it is finished. Please update this afterwards. Thanks!

dchung0741 commented on notebooks/3.3_Quri_Parts.ipynb:
```diff
@@ -0,0 +1,381 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    " pip install \"quri-parts[qulacs]\" \n"
```
I think a "!" is missing here

essafik replied:
This is only necessary if it's run on Google Colab. But maybe we can add a comments about it.

dchung0741 commented on notebooks/3.3_Quri_Parts.ipynb:
```diff
@@ -0,0 +1,381 @@
+    " QuantumGate(\"CNOT\", target_indices=(1,), control_indices=(2,)) \n",
+    "] \n",
+    "for gate in gates: \n",
+    "print(gate)"
```
I'm not sure if this is due to some issue with jupyter notebooks, but the indentation is gone when I view it on my own computer. (It looks like the following on my computer)

![截圖 2023-11-07 下午5 31 11](https://github.com/qulacs/quantum-native-dojo-en/assets/30490924/47b42dc1-cf09-4a97-ad63-d91e449d6c5f)

Could you check this again?

essafik replied:
Will fix it. Thanks

dchung0741 commented on notebooks/3.3_Quri_Parts.ipynb:
```diff
@@ -0,0 +1,381 @@
+    "] \n",
+    "\n",
+    "for gate in gates: \n",
+    "print(gate)"
```
![截圖 2023-11-07 下午5 32 57](https://github.com/qulacs/quantum-native-dojo-en/assets/30490924/4552b6a1-d220-4d0a-93c0-e393677b3051)
Same problem with indentation. Please check again

dchung0741 commented on notebooks/3.3_Quri_Parts.ipynb:
```diff
@@ -0,0 +1,381 @@
+   "source": [
+    "### Operators:\n",
+    "An operator is defined as a set of pairs of a Pauli label and its complex coefficient. For example, an operator:<br>",
+    "$(0.5+0.5j)*X0 Y1 + 0.2j*Z0 Z2 + (0.3+0.4j)*I$",
```
I think it would be more aesthetically pleasing if we use subscripts for the qubit index in the equation.

dchung0741 commented on notebooks/3.3_Quri_Parts.ipynb:
```diff
@@ -0,0 +1,381 @@
+    }
+   ],
+   "source": [
+    "from quri_parts.core.operator import Operator \n",
+    "op = Operator({ \n",
+    "   pauli_label(\"X0 Y1\"): 0.5 + 0.5j, \n",
+    "   pauli_label(\"Z0 Z2\"): 0.2j,\n",
+    "   PAULI_IDENTITY: 0.3 + 0.4j,\n",
+    "}) \n",
+    "print(op)"
```
Missing imports `pauli_label` and `PAULI_IDENTITY`

dchung0741 reviewed (commented):
Thanks for this implementation! I have left some comments (Most of them are about formatting).
A couple of things that can be helpful if they are in the first version of this tutorial as I use them every where in my PRs:
- The 3 types of states in QURI Parts: `GeneralCircuitQuantumState`, `ComputationalBasisState` and `QuantumStateVector`
- The `run_circuit` and `evaluate_state_to_vector` functions quri_parts.qulacs.simulator.
- The `get_sparse_matrix` function in quri_parts.core.operator
- Parametric circuits: `UnboundParametricQuantumCircuit`, `LinearMappedUnboundParametricQuantumCircuit`

Please consider adding them. Thanks!

qalgos committed:
Create 6.4_parameter shift rule

qalgos committed:
Update 6.4_parameter shift rule

qalgos committed:
Rename 6.4_parameter shift rule to 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Update 6.4_parameter shift rule.ipynb

qalgos committed:
Create 10.0_Complexity Theory in a Nutshell

qalgos committed:
Delete notebooks/10.0_Complexity Theory in a Nutshell

qalgos committed:
Create 10_Complexity Theory in a Nutshell

qalgos committed:
Rename 10_Complexity Theory in a Nutshell to 10.1_Complexity Theory in a Nutshell

qalgos committed:
Rename 10.1_Complexity Theory in a Nutshell to z10_Complexity Theory in a Nutshell

qalgos committed:
Rename z10_Complexity Theory in a Nutshell to q10.0_Complexity Theory in a Nutshell

qalgos committed:
Update q10.0_Complexity Theory in a Nutshell

qalgos committed:
Rename q10.0_Complexity Theory in a Nutshell to q10.0_Complexity Theory in a Nutshell.ipynb

qalgos committed:
Update q10.0_Complexity Theory in a Nutshell.ipynb

qalgos committed:
Update q10.0_Complexity Theory in a Nutshell.ipynb

qalgos committed:
Create 4.3_Block encoding.ipynb

qalgos committed:
Update q10.0_Complexity Theory in a Nutshell.ipynb

qalgos committed:
Update q10.0_Complexity Theory in a Nutshell.ipynb

qalgos committed:
Update q10.0_Complexity Theory in a Nutshell.ipynb

qalgos committed:
Update q10.0_Complexity Theory in a Nutshell.ipynb

essafik commented:
> * The 3 types of states in QURI Parts: `GeneralCircuitQuantumState`, `ComputationalBasisState` and `QuantumStateVector`
> * The `run_circuit` and `evaluate_state_to_vector` functions quri_parts.qulacs.simulator.
> * The `get_sparse_matrix` function in quri_parts.core.operator
> * Parametric circuits: `UnboundParametricQuantumCircuit`, `LinearMappedUnboundParametricQuantumCircuit`

Do you want this added just in the introduction or also sho

essafik committed:
Add files via upload

Added Quantum states. But still need to finish this and add example for QuantumStateVector
